tests/testsuite/update.rs

```diff
@@ -1659,3 +1659,56 @@ fn update_with_missing_feature() {
         )
         .run();
 }
+
+#[cargo_test]
+fn update_incompatible() {
+    Package::new("compatible", "1.0.0").publish();
+    Package::new("incompatible", "1.0.0").publish();
+    Package::new("pinned", "1.0.0").publish();
```

Comment:
Also renamed

Reply:
We should also have a `ws_incompatible` to demonstrate workspace inherited dependencies get updated

Reply:
Added these cases and more.